src/components/PeoplePage.tsx

```diff
@@ -1,29 +1,163 @@
 import { PeopleFilters } from './PeopleFilters';
 import { Loader } from './Loader';
 import { PeopleTable } from './PeopleTable';
+import { useEffect, useMemo, useState } from 'react';
+import { Person } from '../types';
+import { getPeople } from '../api';
+import { useParams, useSearchParams } from 'react-router-dom';
+import { PeopleMap } from '../types/PeopleMap';
+
+const getCentury = (year: number): number => {
+  return Math.ceil(year / 100);
+};
 
 export const PeoplePage = () => {
+  const [isLoading, setIsLoading] = useState(true);
+  const [people, setPeople] = useState<Person[]>([]);
+  const [error, setError] = useState<string | null>(null);
+  const { slug: selectedPersonSlug } = useParams();
+  const [searchParams, setSearchParams] = useSearchParams();
+  const sex = searchParams.get('sex');
+  const centuries = searchParams.getAll('centuries');
+  const query = searchParams.get('query');
+  const sortColumn = searchParams.get('sort') as keyof Person | null;
+  const sortDirection = searchParams.get('order') === 'desc';
+
+  useEffect(() => {
+    setIsLoading(true);
+    getPeople()
+      .then(fetchedPeople => {
+        setPeople(
+          fetchedPeople.map(person => ({
+            ...person,
+            century: person.born ? getCentury(person.born) : undefined,
+          })),
+        );
+      })
+      .catch(() => {
+        setError('Something went wrong');
+      })
+      .finally(() => {
+        setIsLoading(false);
+      });
+  }, []);
+
+  const filteredPeople = useMemo(() => {
+    let filtered = people;
+    const centuriesAsNumbers = centuries.map(Number);
+
+    if (sex) {
+      filtered = filtered.filter(person => person.sex === sex);
+    }
+
+    if (centuries.length > 0) {
+      filtered = filtered.filter(
+        person => person.century && centuriesAsNumbers.includes(person.century),
+      );
+    }
+
+    if (query) {
+      filtered = filtered.filter(person =>
+        person.name.toLowerCase().includes(query.toLowerCase()),
+      );
+    }
+
+    return filtered;
+  }, [people, centuries, query, sex]);
+
+  const sortedFilteredPeople = useMemo(() => {
+    if (!sortColumn) {
+      return filteredPeople;
+    }
+
+    return [...filteredPeople].sort((a, b) => {
+      const valueA = a[sortColumn];
+      const valueB = b[sortColumn];
+
+      if (valueA === undefined) {
+        return 1;
+      }
+
+      if (valueB === undefined) {
+        return -1;
+      }
+
+      if (valueA < valueB) {
+        return sortDirection ? 1 : -1;
+      }
+
+      if (valueA > valueB) {
+        return sortDirection ? -1 : 1;
+      }
+
+      return 0;
+    });
+  }, [filteredPeople, sortColumn, sortDirection]);
+  const handleSearchChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const newParams = new URLSearchParams(searchParams);
+
+    newParams.set('query', event.target.value);
+    setSearchParams(newParams);
```

Comment:
Not necessary, but maybe you could use something similar to the approach in setState:
```
setSearchParams(params => {
    params.set('query', event.target.value);
    return params;
});